lib/soapy_cake/request.rb

```diff
@@ -38,6 +38,10 @@ def to_s
       "#{role}:#{service}:#{method}:#{version} #{opts.to_json}"
     end
 
+    def read_only?
+      (API_CONFIG.dig('read_only', role, service) || []).include?(method)
```

Comment:
Right, `try :[].try :[].try :[]` looks ugly, while `dig` does the same without raising suspicion.